docs/markdown/Fs-module.md

```diff
@@ -194,3 +194,20 @@ Returns the last component of the path, dropping the last part of the suffix
 fs.stem('foo/bar/baz.dll')  # baz
 fs.stem('foo/bar/baz.dll.a')  # baz.dll
 ```
+
+### relative_to
+
+*since 0.56.0*
+
+Given two absolute paths, returns a version of the first path relative to the second.
+If a path is given as the 'within' argument is specified, the function will return the
+first path unchanged if it is not inside 'within'.
+
+```meson
+fs.relative_to('/prefix/lib', '/prefix/bin')  # '../lib'
+fs.relative_to('/prefix/lib/foo', '/prefix')  # 'lib/foo'
+fs.relative_to('/usr/lib/foo', '/usr/bin', within: '/usr')  # '../lib/foo'
+fs.relative_to('/usr/lib/foo', '/usr/bin', within: '/var')  # '/usr/lib/foo'
```

Comment:
This should also specify what happens on Windows paths that are on different drives:

```meson
fs.relative_to('c:\tmp', 'd:\something') # ??? (should return the latter path maybe?)

Reply:
I have improved Windows support in relative_to based on your comment.

Since proper absolute/relative path computation is system dependent (e.g. pathlib's PurePath does not detect absolute Windows paths on a Linux system), I've added a `native` argument, so the user can choose if he/she needs the relative path for the host or build system.

Besides, I have added tests for windows paths and clarified possible error messages.

This behaviour is now documented:

    fs.relative_to('c:\\proj1\\foo', 'd:\\proj1\\bar')  # ERROR, since the relative path does not exist. Consider using "within".
    fs.relative_to('c:\\proj1\\foo', 'd:\\proj1\\bar', within: 'd:\\')  # 'c:\\proj1\\foo' ,  "within" allows us to get an absolute path if we are not within that directory:

In my opinion it makes sense that `relative_to` always returns a relative path unless we explicitly allow for absolute paths to be returned (using `within`).

Reply:
As a matter of curiosity -- what's the use case for doing anything other than erroring out when trying to compute the relative path between two different drives?

Python considers this to be a ValueError, which doesn't seem illogical.

Reply:
I guess sometimes one would prefer to have a relative path if possible, but would accept an absolute path otherwise rather than an error